src/video/vga_state.c

```diff
@@ -47,9 +48,14 @@ void vga_state_pop_palette(void) {
     damage_set_range(&state.dmg_base, 0, 255);
 }
 
-void vga_state_render(void) {
+void vga_state_render(game_state *gs) {
+    unsigned int old_count = state.transformer_count;
+    // recalculate transformers stack
+    state.transformer_count = 0;
+    game_state_palette_transform(gs);
```

Comment:
vga_state should not reach into other subsystems. This function should be called from engine.c instead.